app/models/plant_line.rb

```diff
@@ -30,10 +30,14 @@ def self.grid_data(filter)
       'data_owned_by',
       'organisation'
 
-    query = where(plant_line_name: filter[:plant_line_names]) if filter[:plant_line_names].present?
+    query = where(filter[:query]) if filter[:query].present?
     query = where('plant_line_name ILIKE ?', "%#{filter[:search]}%") if filter[:search].present?
     query ||= none
 
+    filter[:query].each do |k,_|
```

Comment:
Did you introduce it because you will be adding querying by many other fields? I do not see anything wrong with it provided you make sure `filter[:query]` is limited to a `Hash`, otherwise it allows SQL injection (I know it is checked in the controller, I wonder if it should not be checked here too).